src/lib/web-server.js

```diff
@@ -9,6 +10,17 @@ web_server_app.get('/', (req, res) => {
     );
 });
 
+web_server_app.get('/normalised_data/all', async (req, res) => {
+    const query = new rpde_query(req.query.afterTimestamp, req.query.afterId, req.query.limit)
+    await query.run();
+    res.json(
+        {
+            "next": "/normalised_data/all?" + query.get_next_get_params_string(),
+            "items": query.get_data()
```

Comment:
this could get into a bit of race condition, couldn't run just return the data?

Reply:
How? All processing is done in the run() function with one SQL statement. The get_next_get_params_string and get_data only return data stored on the object, and don't do any DB work of their own.